tests/functional_tests/ffront_tests/test_execution.py

```diff
@@ -848,6 +812,118 @@ def solve_tridiag(
+    def ternary_field_op(
+        a: Field[[IDim], float], b: Field[[IDim], float], left: float, right: float
+    ) -> Field[[IDim], float]:
+        return a if left < right else b
+
+    ternary_field_op(a, b, left, right, out=out, offset_provider={})
+    e = np.asarray(a) + np.asarray(b) if left < right else np.asarray(b)
+    np.allclose(e, out)
+
+    @field_operator
+    def ternary_field_op_scalars(left: float, right: float) -> Field[[IDim], float]:
+        return broadcast(3.0, (IDim,)) if left < right else broadcast(4.0, (IDim,))
+
+    ternary_field_op_scalars(left, right, out=out, offset_provider={})
+    e = np.full(e.shape, 3.0) if left < right else e
+    np.allclose(e, out)
+
+
+def test_ternary_operator_tuple():
+
+    size = 10
+    a = np_as_located_field(IDim)(np.ones((size,)))
+    b = np_as_located_field(IDim)(2 * np.ones((size,)))
+    out_1 = np_as_located_field(IDim)(np.zeros((size,)))
+    out_2 = np_as_located_field(IDim)(np.zeros((size,)))
+
+    left = 2.0
+    right = 3.0
+
+    @field_operator
+    def ternary_field_op(
+        a: Field[[IDim], float], b: Field[[IDim], float], left: float, right: float
+    ) -> tuple[Field[[IDim], float], Field[[IDim], float]]:
+        return (a, b) if left < right else (b, a)
+
+    # TODO(tehrengruber): directly call field operator when the generated programs support `out` being a tuple
+    @program
+    def ternary_field(
+        a: Field[[IDim], float],
+        b: Field[[IDim], float],
+        left: float,
+        right: float,
+        out_1: Field[[IDim], float],
+        out_2: Field[[IDim], float],
+    ):
+        ternary_field_op(a, b, left, right, out=(out_1, out_2))
+
+    ternary_field(a, b, left, right, out_1, out_2, offset_provider={})
+
+    e, f = (np.asarray(a), np.asarray(b)) if 2 < 3 else (np.asarray(b), np.asarray(a))
+    np.allclose(e, out_1)
+    np.allclose(f, out_2)
+
+
+def test_ternary_builtin_neighbor_sum(reduction_setup):
+    rs = reduction_setup
+    Edge = rs.Edge
+    Vertex = rs.Vertex
+    V2EDim = rs.V2EDim
+    V2E = rs.V2E
+
+    num_vertices = rs.num_vertices
+    num_edges = rs.num_edges
+
+    a = np_as_located_field(Edge)(np.ones((num_edges,)))
+    b = np_as_located_field(Edge)(2 * np.ones((num_edges,)))
+    out = np_as_located_field(Vertex)(np.zeros((num_vertices,)))
+
+    @field_operator
+    def ternary_reduce(a: Field[[Edge], float], b: Field[[Edge], float]) -> Field[[Vertex], float]:
+        out = neighbor_sum(b(V2E) if 2 < 3 else a(V2E), axis=V2EDim)
+        return out
+
+    ternary_reduce(a, b, out=out, offset_provider=rs.offset_provider)
+
+    expected = (
+        np.sum(np.asarray(b)[rs.v2e_table], axis=1)
+        if 2 < 3
+        else np.sum(np.asarray(a)[rs.v2e_table], axis=1)
+    )
+
+    assert np.allclose(expected, out)
+
+
+def test_ternary_scan():
+    KDim = Dimension("K", kind=DimensionKind.VERTICAL)
+    size = 10
+    init = 1.0
+    a = np_as_located_field(KDim)(np.ones((size,)))
```

Comment:
```suggestion
    init = 0.0
    a = np_as_located_field(KDim)(4*np.ones((size,)))
```
What pattern is chosen here doesn't matter that much, but the idea why I asked for this test to be adopted was to have a test where the condition changes depending on the k-level on which the scan pass, i.e. the scalar function, is executed. Result should be `0, 1, 2, 3, 4, 5, 5, 5, 5, 5`.